classes/02class/exercises/c02-network07/rodrigovcesar/README.md

```diff
@@ -0,0 +1,201 @@
+{
+    "Return": true
+}
+
+## Private
+
+aws ec2 create-route --route-table-id rtb-013716806215ecf49 --destination-cidr-block 0.0.0.0/0 --nat-gateway-id nat-089351075b8a819ed
+{
+    "Return": true
+}
+
+
+
+```
+
+- Answer the following questions:
+  - How did you accomplish both instructions?
+    ```
+    Mainly through the AWS docs, also a bit on Plural Sight to understand better the concepts.
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+    ```
+  Because it represents the least specific route, whenever a destination is not explicitly known to the route table it is redirected to the IGW under the 0.0.0.0 address (representing an external resource, all the IPV4 addresses if I am not mistaken in terminology here).
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+    ```
+  It's the same as the previous answer, it is an external route not known to the route table, so it is redirected to this address, with the difference we are redirecting the traffic to the NAT since this is a private instance for outbound traffic.
+    ```
+    
+  - What is the difference of IGW and NGW?
+    ```
+    The Internet Gateway is a VPC component and represents a connection between the internet and the VPC itself. Only one can be associated with each VPC.
+
+    NAT gateway enable instances in a private subnet to connect to the internet or other AWS services, but prevent the internet from initiating a connection with those instances. 
+    
+    Instances in a private subnet that want to access the Internet can have their Internet-bound traffic forwarded to the NAT Instance via a Route Table configuration. The NAT Instance will then make the request to the Internet (since it is in a Public Subnet) and the response will be forwarded back to the private instance. (source https://www.edureka.co/community/2149/aws-vpc-what-is-the-difference-between-internet-gateway-nat).
+
+    ```
+    
+  - Can you delete the destination route to your VPC network? Why?
+    ```
+    - No, because it's a local route, otherwise it would cause disruption on the communication between the subnets and the VPC.
+    ```
+    
+  - What happens if you do not associate your route table with any subnets?
+    ```
+    All VPCs have one main table route (RT) by default, controlling the routing for all subnets that are not explicitly associated with any RT.  
+    
+    Every subnet, when created and not explicitly assigned with any RT, is automatically associated with the main route table from the VPC - an implicity association.   
+    
+    A subnet can only be associated with one RT at a time, but you can associate multiple subnets with the same subnet RT.
+    
+    In this case, from what I understand,  creating a custom RT without any association will not produce any effect at all if you do not have an explicit association.     
```

Comment:
Correct, the route will stay there unused (unless you make it as default route, which could get the unassociated subnets)